test/unit/StakingPool/calculatePremium.js

```diff
@@ -0,0 +1,611 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+const { parseEther } = ethers.utils;
+const { BigNumber } = ethers;
+const {
+  calculateSurgePremiums,
+  calculateBasePrice,
+  divCeil,
+  calculateFixedPricePremium,
+  calculatePriceBump,
+} = require('./helpers');
+const { daysToSeconds } = require('../../../lib/helpers');
+
+describe('calculatePremium', function () {
+  // If an arithmetic operation results in underflow or overflow outside of an unchecked { ... } block.
+  const UNDER_OR_OVERFLOW = 0x11;
+  // If you divide or modulo by zero (e.g. 5 / 0 or 23 % 0).
+  const DIVIDE_BY_ZERO = 0x12;
+
+  const stakedProductTemplate = {
+    lastEffectiveWeight: BigNumber.from(50),
+    targetWeight: BigNumber.from(70), // 70%
+    targetPrice: BigNumber.from(200), // 2%
+    nextPrice: BigNumber.from(200), // 2%
+    nextPriceUpdateTime: BigNumber.from(0),
+  };
+
+  it('should calculate premium on multiple cover buys over time, based on pre-defined numbers', async function () {
+    const { stakingPool } = this;
+    let { timestamp } = await ethers.provider.getBlock('latest');
+    timestamp = BigNumber.from(timestamp);
+    const { NXM_PER_ALLOCATION_UNIT, PRICE_BUMP_RATIO, PRICE_CHANGE_PER_DAY, INITIAL_PRICE_DENOMINATOR } = this.config;
+    let stakedProduct = { ...stakedProductTemplate, nextPriceUpdateTime: timestamp };
+    const period = daysToSeconds(1);
+    const totalCapacity = divCeil(parseEther('50000'), NXM_PER_ALLOCATION_UNIT);
+
+    // 1st cover buy
+    {
+      const coverAmountRaw = parseEther('2400');
+      const coverAmount = divCeil(coverAmountRaw, NXM_PER_ALLOCATION_UNIT);
+      const initialCapacityUsed = BigNumber.from(0);
+      const basePrice = calculateBasePrice(timestamp, stakedProduct, PRICE_CHANGE_PER_DAY);
+      expect(basePrice).to.be.equal(200);
+      const priceBump = calculatePriceBump(coverAmount, PRICE_BUMP_RATIO, totalCapacity);
+      expect(priceBump).to.be.equal(96);
+
+      const expectedPremiun = calculateFixedPricePremium(
+        coverAmount,
+        period,
+        basePrice,
+        NXM_PER_ALLOCATION_UNIT,
+        INITIAL_PRICE_DENOMINATOR,
+      );
+      const [premium, product] = await stakingPool.calculatePremium(
+        stakedProduct,
+        period,
+        coverAmount,
+        initialCapacityUsed,
+        totalCapacity,
+        basePrice,
```

Comment:
basePrice here is passed for what the function needs as targetPrice

Was this the intention here?

Reply:
Oh nice catch. Should be targetPrice instead..